aidants_connect_web/views/FC_as_FS.py

```diff
@@ -115,16 +115,10 @@ def fc_callback(request):
         log.info("403: The connection has expired.")
         return HttpResponseForbidden()
 
-    try:
-        usager_sub = generate_sha256_hash(
-            f"{decoded_token['sub']}{settings.FC_AS_FI_HASH_SALT}".encode()
-        )
-        usager = Usager.objects.get(sub=usager_sub)
-    except Usager.DoesNotExist:
-        usager, error = get_user_info(fc_base, connection.access_token)
-        if error:
-            messages.error(request, error)
-            return redirect("dashboard")
+    usager, error = get_user_info(connection)
```

Comment:
Petit commentaire "stylistique" qui dépasse clairement le _scope_ de cette PR, donc pas bloquant :)

Le _pattern_ qui consiste à renvoyer un résultat + un éventuel code d'erreur, comme ici, est certes rendu possible par la souplesse de Python, mais n'est cependant pas considéré comme très _pythonique_ ; pour des raisons de lisibilité et expressivité du code, il est souvent conseillé d'utiliser des exceptions pour gérer ce type de _flow_.

Quelque chose du genre :
```python
def get_user_info(connection):
    result = do_something()  # this may raise MyCustomException
    return result

try:
    user_info = get_user_info(connection)
except MyCustomException:
    handle_this_specific_case()
```

C'est un _pattern_ qu'on retrouve souvent dans Django, par exemple [au niveau de la validation de formulaires](https://docs.djangoproject.com/fr/3.0/ref/forms/validation/), avec l'utilisation de l'exception `ValidationError`.

/cc @Anna-Livia :)

Reply:
Pour aller un peu plus loin à ce sujet, je te/vous propose la lecture de [cet article](https://jeffknupp.com/blog/2013/02/06/write-cleaner-python-use-exceptions/) :)

Reply:
very true, je trouve ca bizarre aussi qu'une méthode renvoie un tuple. je note aussi dans la todo refacto :)

Reply:
Il y a des circonstances où c'est intéressant, par exemple dans le cas de `get_or_create` justement, qui te renvoie le résultat _et_ une "métadonnée" sur celui-ci (l'objet existait-il déjà ou pas ?), mais dans le cas des erreurs la plupart du temps le code est plus lisible avec des exceptions :)